tests/Unit/Products/ProductUnitTest.php

```diff
@@ -3,39 +3,86 @@
 namespace Tests\Unit\Products;
 
 use Illuminate\Http\UploadedFile;
-use IndieHD\Velkart\Product\Product;
+use IndieHD\Velkart\Product\Contracts\ProductRepositoryContract;
 use IndieHD\Velkart\Tests\TestCase;
 
 class ProductUnitTest extends TestCase
 {
+    protected $repo;
+
+    public function setUp(): void
+    {
+        parent::setUp();
+
+        $this->repo = resolve(ProductRepositoryContract::class);
+    }
+
     /** @test */
     public function it_can_create_a_product()
     {
-        $product = 'apple';
-
-        $cover = UploadedFile::fake()->image('file.png', 600, 600);
+        $product = 'Samsung Galaxy S10';
 
         $params = [
-            'sku' => $this->faker->numberBetween(1111111, 999999),
+            'sku' => $this->faker->unique()->numberBetween(1111111, 999999),
             'name' => $product,
             'slug' => str_slug($product),
             'description' => $this->faker->paragraph,
-            'cover' => $cover,
+            'cover' => UploadedFile::fake()->image('product.png', 600, 600),
             'quantity' => 10,
             'price' => 9.95,
             'status' => 1,
         ];
 
-        $product = factory(Product::class)->make($params);
-
-        $this->assertInstanceOf(Product::class, $product);
-        $this->assertEquals($params['sku'], $product->sku);
-        $this->assertEquals($params['name'], $product->name);
-        $this->assertEquals($params['slug'], $product->slug);
-        $this->assertEquals($params['description'], $product->description);
-        $this->assertEquals($params['cover'], $product->cover);
-        $this->assertEquals($params['quantity'], $product->quantity);
-        $this->assertEquals($params['price'], $product->price);
-        $this->assertEquals($params['status'], $product->status);
+        $created = factory($this->repo->modelClass())->create($params);
+
+        $this->assertInstanceOf($this->repo->modelClass(), $created);
+        $this->assertEquals($params['sku'], $created->sku);
+        $this->assertEquals($params['name'], $created->name);
+        $this->assertEquals($params['slug'], $created->slug);
+        $this->assertEquals($params['description'], $created->description);
+        $this->assertEquals($params['cover'], $created->cover);
+        $this->assertEquals($params['quantity'], $created->quantity);
+        $this->assertEquals($params['price'], $created->price);
+        $this->assertEquals($params['status'], $created->status);
+    }
+
+    /** @test */
+    public function it_can_list_all_the_products()
```

Comment:
These test method names should adhere to PSR-2, or they'll throw warnings when we add GrumPHP.